components/tooltip/Tooltip.js

```diff
@@ -0,0 +1,271 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import Portal from '../hoc/Portal';
+import { getViewport } from '../utils/utils';
+import events from '../utils/events';
+import theme from './theme.css';
+
+const POSITION = {
+  BOTTOM: 'bottom',
+  HORIZONTAL: 'horizontal',
+  LEFT: 'left',
+  RIGHT: 'right',
+  TOP: 'top',
+  VERTICAL: 'vertical',
+};
+
+const defaults = {
+  className: '',
+  color: 'white',
+  delay: 0,
+  hideOnClick: true,
+  icon: null,
+  passthrough: true,
+  showOnClick: false,
+  size: 'medium',
+  position: POSITION.VERTICAL,
+};
+
+const tooltipFactory = (options = {}) => {
+  const {
+    className: defaultClassName,
+    color: defaultColor,
+    delay: defaultDelay,
+    hideOnClick: defaultHideOnClick,
+    icon: defaultIcon,
+    showOnClick: defaultShowOnClick,
+    size: defaultSize,
+    passthrough: defaultPassthrough,
+    position: defaultPosition,
+  } = { ...defaults, ...options };
+
+  return ComposedComponent => {
+    class TooltippedComponent extends Component {
+      static propTypes = {
+        children: PropTypes.node,
+        className: PropTypes.string,
+        onClick: PropTypes.func,
+        onMouseEnter: PropTypes.func,
+        onMouseLeave: PropTypes.func,
+        tooltip: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+        tooltipColor: PropTypes.oneOf(['white', 'neutral', 'mint', 'violet', 'ruby', 'gold', 'aqua', 'inverse']),
+        tooltipDelay: PropTypes.number,
+        tooltipHideOnClick: PropTypes.bool,
+        tooltipIcon: PropTypes.element,
+        tooltipPosition: PropTypes.oneOf(Object.keys(POSITION).map(key => POSITION[key])),
+        tooltipShowOnClick: PropTypes.bool,
+        tooltipSize: PropTypes.oneOf(['small', 'medium']),
+      };
+
+      static defaultProps = {
+        className: defaultClassName,
+        tooltipColor: defaultColor,
+        tooltipDelay: defaultDelay,
+        tooltipHideOnClick: defaultHideOnClick,
+        tooltipIcon: defaultIcon,
+        tooltipPosition: defaultPosition,
+        tooltipShowOnClick: defaultShowOnClick,
+        tooltipSize: defaultSize,
+      };
+
+      state = {
+        active: false,
```

Comment:
**PP** Prefer using `isCondition` signature for boolean values, eg. `isActive`.

Reply:
As we recently discussed, we'd prefer to pass the props without the is prefix. But, functions that return a boolean value we should always prefix with is, has, ...